pkg/nfd-master/nfd-master.go

```diff
@@ -580,6 +598,11 @@ func (m *nfdMaster) filterFeatureLabels(labels Labels, features *nfdv1alpha1.Fea
 		}
 	}
 
+	if m.config.Restrictions.MaxLabelsPerCR > 0 && len(outLabels) > m.config.Restrictions.MaxLabelsPerCR {
```

Comment:
Ummm, what if instead of skipping we trim the `outLabels` to the MAX number? If the user took time to specify a max number, they expect a min number of labels, not 0. debugging 0 vs "missing" labels is a diff scenario.

Reply:
I think it's an area for discussion here, since we already discussed in the comments section https://github.com/kubernetes-sigs/node-feature-discovery/issues/1380#issuecomment-1884484022
wdyt @marquiz

Reply:
IMO it should be "transactional" i.e. all or nothing. If we trim to MAX if user asks for MAX+1 it's sorta random what gets left out. Plus very hard to spot the problem if e.g. only one of the labels is missing.

Reply:
Hmm, looking at this now it's not "PerCR" but just maximum number of labels. So this (and the others) should be renamed.

```suggestion
	if m.config.Restrictions.MaxLabels > 0 && len(outLabels) > m.config.Restrictions.MaxLabels {
```